pkg/handler/processor/process/process_test.go

```diff
@@ -498,9 +498,17 @@ func Test_SimpleDocProcessTest(t *testing.T) {
 	},
 	}
 
-	logrus.SetLevel(logrus.DebugLevel)
+	zapLogger, _ := zap.NewDevelopmentConfig().Build()
+
+	// flushes buffer, if any
+	defer func() {
+		// intentionally ignoring error here, see https://github.com/uber-go/zap/issues/328
+		_ = zapLogger.Sync()
+	}()
+	logger := zapLogger.Sugar()
+
 	// Register
-	RegisterDocumentProcessor(&simpledoc.SimpleDocProc{}, simpledoc.SimpleDocType)
+	RegisterDocumentProcessor(&simpledoc.SimpleDocProc{}, simpledoc.SimpleDocType, logger)
```

Comment:
have this not take in a logger and return error instead? With an err that can be compared to check if it is a warning for a overwrite of a registration.

Reply:
Addresed ✅